src/bin/stratisd.rs

```diff
@@ -83,11 +83,22 @@ fn handle_udev_add(event: &libudev::Event) -> Option<(Device, PathBuf)> {
 /// To ensure only one instance of stratisd runs at a time, acquire an
 /// exclusive lock. Return an error if lock attempt fails.
 fn trylock_pid_file() -> StratisResult<File> {
-    let mut f = OpenOptions::new()
+    let mut f = match OpenOptions::new()
         .read(true)
         .write(true)
         .create(true)
-        .open(STRATISD_PID_PATH)?;
+        .open(STRATISD_PID_PATH)
+    {
+        Ok(f) => f,
+        Err(ref e) if e.kind() == ErrorKind::PermissionDenied => {
+            println!(
+                "Permission denied while accessing lock file.\n\
+                 Please make sure to run this command as root."
+            );
+            exit(1);
```

Comment:
Instead of exiting please return a ```StratisError``` with the text of the error message. The caller will immediately exit w/ that error, see below.